design/ceph/multus-network.md

```diff
@@ -0,0 +1,134 @@
+
+## Multus supported configuration
+
+### Interface type
+
+As part of the CNI spec, Multus supports several [interface types](https://github.com/containernetworking/plugins#main-interface-creating).
+Rook will naturally support any of them as they don't fundamentally change the working behavior.
+
+### IPAM type
+
+This is where things get more complex.
+Currently there are [three different IPAM](https://github.com/containernetworking/plugins#ipam-ip-address-allocation) solutions available.
+
+As part of our research we have found that the following IPAM types are not good candidates:
+
+- host-local: Maintains a local database of allocated IPs, this only works on a per host basis, so not suitable for a distributed environment since we will end up with IP collision
+To fix this, the [whereabouts project](https://github.com/dougbtv/whereabouts) looks promising but is not officially supported.
+- static: Allocate a static IPv4/IPv6 addresses to container and it's useful in debugging purpose.
+This cannot at scale because this means we will have to allocate IPs for **all** the daemon so it's not scalable.
+
+You can find a more detailed analysis at the end of the document in the [rejected proposal section](#rejected-proposals).
+
+## Ceph daemons implementation challenges
+
+### Monitors and OSDs
+
+The Ceph monitors only need access to the public network.
+The OSDs needs access to both public and cluster networks.
+
+Monitors requirements so far are the following:
+
+- Predictable IP addresses
+- Keep the same IP address for the life time of the deployment (IP should survive a restart)
+
+### RGW implementation
+
+Only need access to the public network.
+They use service IP with a load-balancer so we need to be careful.
+
+### MDS/RBD-MIRROR/NFS implementation
+
+Only need access to the public network.
+Nothing to do in particular since they don't use any service IPs.
+
+### CSI pods
+
+We can add annotations to these pods and they can reach out the Ceph public network, then the driver will expose the block or the filesystem normally.
+
+## Accepted proposal
+
+So far, the team has decided to go with the [whereabouts](https://github.com/dougbtv/whereabouts) IPAM.
+It's an IP Address Management (IPAM) CNI plugin that assigns IP addresses cluster-wide.
+If you need a way to assign IP addresses dynamically across your cluster -- Whereabouts is the tool for you. If you've found that you like how the host-local CNI plugin works, but, you need something that works across all the nodes in your cluster (host-local only knows how to assign IPs to pods on the same node) -- Whereabouts is just what you're looking for.
+Whereabouts can be used for both IPv4 & IPv6 addressing.
+
+It is under active development and is not ready but ultimately will allow to:
+
+- Have static IP addresses distributed across the cluster
+- These IPs will survive any deployment restart
+- The allocation will be done by whereabouts and not by Rook
+
+/!\
```

Comment:
Is this supposed to be rendered as a warning in markdown? Looks like it is not rendered

Reply:
No, just a highlight :)